src/argprocessing.cpp

```diff
@@ -222,6 +222,14 @@ process_arg(Context& ctx,
     return nullopt;
   }
 
+  for (const std::string& option : ctx.ignore_options) {
+    if (option == args[i] || (Util::ends_with(option, "*") &&
+          Util::starts_with(args[i], option.substr(0, option.length() - 1)))) {
```

Comment:
How about extract these two lines of the if statement in a separate function? Maybe the loop as well. Something like `bool optionToBeIgnoredForCache()`?
It doesn't seem like much, but then the if statement would be self documented:
```
if(optionToBeIgnoredForCache(ctx.ignore_options, args[i])) {
      cc_log("Skipping argument: %s", args[i].c_str());
      return nullopt;
}
```

Reply:
Good idea. I’ll do that.